janusgraph-core/src/main/java/org/janusgraph/graphdb/vertices/StandardVertex.java

```diff
@@ -33,16 +33,19 @@
 
 public class StandardVertex extends AbstractVertex {
 
-    private byte lifecycle;
+    private Object lifecycleMutex = new Object();
```

Comment:
I think this should be final.

Reply:
It's marked as final. Were you going to suggest somewhere I could add a unit test? Please give me some guidance on how you would like it to be tested as this is a timing issue so not sure how to reproduce it consistently with a unit test.

I'd like to get this checked in, because of this error I have to do a workaround which adds a performance penalty to my code.

Reply:
Couldn't you add a test that performs the steps you described in the original issue and then execute that code like 10 times? Would that result in the deadlock more or less consistently without this fix?

Reply:
@FlorianHockmann I think that it will be a new flaky test which will fail too often than. `updateLifeCycle` executes too fast so 10 times won't be enough for sure. There is nothing to mock in `updateLifeCycle` so, the only way to test it I see is to spy on some method and make the logic there which waits some trigger.
I suggest to spy on `remove` method because it is more complicated then `removeRelation` method. I will provide an example below in comments. Please check if you agree with it.